source/val/validation_state.cpp

```diff
@@ -187,6 +191,21 @@ ValidationState_t::ValidationState_t(const spv_const_context ctx,
     default:
       break;
   }
+
+  // Only attempt to count if we have words, otherwise let the other validation
+  // fail and generate an error.
+  if (num_words > 0) {
```

Comment:
I do this work in the constructor because ValidationState is used from both the validator and the stats aggregator. We either do it here or we add methods to do it that we have to remember to call from each place we want to use ValidationState in the future.